cdap-authorization-client/src/main/java/co/cask/cdap/authorization/ACLManagerClient.java

```diff
@@ -0,0 +1,198 @@
+                              ": " + response.getResponseMessage());
+    }
+  }
+
+  /**
+   * Creates an {@link ACLEntry} in a namespace for an object, subject, and a permission.
+   * This allows the subject to access the object for the specified permission.
+   *
+   * <p>
+   * For example, if object is "secretFile", subject is "Bob", and permission is "WRITE", then "Bob"
+   * would be allowed to write to the "secretFile", assuming that what is doing the writing is protecting
+   * the "secretFile" via a call to one of the {@code verifyAuthorized()} or {@code isAuthorized()} calls.
+   * </p>
+   *
+   * @param entry the {@link ACLEntry} to create
+   * @throws java.io.IOException if an error occurred when contacting the authorization service
+   */
+  public void createACL(String namespaceId, ACLEntry entry) throws IOException {
+    HttpRequest request = HttpRequest.post(resolveURL("/v1/acls/namespace/" + namespaceId))
+      .withBody(GSON.toJson(entry)).addHeaders(getHeaders()).build();
+    HttpResponse response = HttpRequests.execute(request);
+
+    if (response.getResponseCode() != HttpURLConnection.HTTP_OK &&
+      response.getResponseCode() != HttpURLConnection.HTTP_NOT_MODIFIED) {
+      throw new IOException("Unexpected response: " + response.getResponseCode() +
+                              ": " + response.getResponseMessage());
+    }
+  }
+
+  /**
+   * Creates an {@link ACLEntry} for the global namespace, subject, and a permission.
+   * This allows the subject to access the object for the specified permission.
+   *
+   * <p>
+   * For example, if object is "secretFile", subject is "Bob", and permission is "WRITE", then "Bob"
+   * would be allowed to write to the "secretFile", assuming that what is doing the writing is protecting
+   * the "secretFile" via a call to one of the {@code verifyAuthorized()} or {@code isAuthorized()} calls.
+   * </p>
+   *
+   * @param entry the {@link ACLEntry} to create
+   * @throws java.io.IOException if an error occurred when contacting the authorization service
+   */
+  public void createGlobalACL(ACLEntry entry) throws IOException {
+    HttpRequest request = HttpRequest.post(resolveURL("/v1/acls/global"))
+      .withBody(GSON.toJson(entry)).addHeaders(getHeaders()).build();
+    HttpResponse response = HttpRequests.execute(request);
+
+    if (response.getResponseCode() != HttpURLConnection.HTTP_OK &&
+      response.getResponseCode() != HttpURLConnection.HTTP_NOT_MODIFIED) {
+      throw new IOException("Unexpected response: " + response.getResponseCode() +
+                              ": " + response.getResponseMessage());
+    }
+  }
+
+  protected URL resolveURL(String path) throws MalformedURLException {
+    return baseURISupplier.get().resolve(path).toURL();
+  }
+
+  private Multimap<String, String> getHeaders() {
+    return headersSupplier == null ? HashMultimap.<String, String>create() : headersSupplier.get();
```

Comment:
Creation of new `Multimap` can be avoided when `headersSupplier` is `null` by using `ImmutableMultimap.of()`